client/src/embedded_support.ts

```diff
@@ -77,18 +88,44 @@ function isPropertyAssignmentToStringOrStringInArray(
     const isCursorInToken = scanner.getStartPos() <= offset &&
         scanner.getStartPos() + scanner.getTokenText().length >= offset;
     if (propertyAssignmentContext && isCursorInToken && isStringToken) {
-      return true;
+      const start = scanner.getTokenPos();
+      const end = start + scanner.getTokenText().length;
+      return {start, end};
     }
 
     lastTokenText = scanner.getTokenText();
     lastToken = token;
     token = scanner.scan();
   }
 
-  return false;
+  return undefined;
 }
 
 function isPropertyAssignmentTerminator(token: ts.SyntaxKind) {
   return token === ts.SyntaxKind.EndOfFileToken || token === ts.SyntaxKind.CommaToken ||
       token === ts.SyntaxKind.SemicolonToken || token === ts.SyntaxKind.CloseBraceToken;
+}
+
+/** Takes a TS file and strips out all non-inline template content */
+export function getHTMLVirtualContent(
+    document: vscode.TextDocument, position: vscode.Position): string {
+  const documentText = document.getText();
+  const offset = document.offsetAt(position);
+  // Replace everything in the document with whitespace
+  let content = documentText.split('\n')
+                    .map(line => {
+                      return ' '.repeat(line.length);
+                    })
+                    .join('\n');
+
+  const templateRegion = getPropertyAssignmentRegion(documentText, offset, ['template']);
```

Comment:
Do we really need this check? Didn't we already check that the cursor is `isInsideInlineTemplateRegion`?

Reply:
Resolved by removing this part completely. It seems as though the html providers are capable of handling the full content of the TS file and interpreting HTML at the position.